### PR TITLE
Refactors the enumeration generation code

### PR DESCRIPTION
**Pull request by JLLeitschuh**
Generator can now generate enumerations for any opencv source files

### CODE REVIEW

File: buildSrc/src/main/java/edu/wpi/gripgenerator/FileParser.java [hunk 1]
```diff
@@ -111,7 +117,61 @@ private static CompilationUnit readFile(URL url) {
...
                 new DefinedMethod("cornerHarris", false, "Mat", "Mat"),
-                new DefinedMethod("cornerEigenValsAndVecs", false, "Mat", "Mat")
+                new DefinedMethod("cornerEigenValsAndVecs", false, "Mat", "Mat"),
+                new DefinedMethod("threshold", false,
+                        new DefinedParamType("Mat"),
+                        new DefinedParamType("Mat"),
+                        new DefinedParamType("double"),
+                        new DefinedParamType("double"),
+                        new DefinedParamType("int").setLiteralDefaultValue("THRESH_BINARY")
+                ).addDescription("Applies a fixed-level threshold to each array element. " +
```
Comment (ThomasJClark) [inline buildSrc/src/main/java/edu/wpi/gripgenerator/FileParser.java]: This description is way too long.  Keep it a line or two, and users can refer to the opencv documentation for more information

**Comment by ThomasJClark** [comment]
Not all enumerations work - `flipCode` still shows up as a number.

**Comment by ThomasJClark** [comment]
`dtype` also shows up as a number, although we can safely get rid of that for now

**Comment by ThomasJClark** [comment]
I keep getting unknown interpolation method errors when I use `INTER_MAX`, `WARP_FILL_OUT`, or `WARP_INVERSE_MAP`.  I think it's generating the wrong values for those

**Comment by JLLeitschuh** [comment]
They aren't the wrong values. I can guarantee that. Do they start working when you try to resize bigger instead of smaller?

**Comment by JLLeitschuh** [comment]
Flip code I have to write by hand and add.

**Comment by JLLeitschuh** [comment]
I have to write a rule to ignore dtype.

**Comment by ThomasJClark** [comment]
> They aren't the wrong values. I can guarantee that. Do they start working when you try to resize bigger instead of smaller?

What?  Where are you getting this information?  The opencv documentation I'm reading contradicts this.  `INTER_MAX` is the number of interpolation modes, and `WARP_FILL_OUTLIERS` and `WARP_INVERSE_MAP` are flags.  The generator is treating them as enum values.

**Comment by ThomasJClark** [comment]
> Flip code I have to write by hand and add.

> I have to write a rule to ignore dtype.

Ok, that's fine.  Can you do these in this PR?

**Comment by JLLeitschuh** [comment]
Hmmm.. Perhaps if an enum is used as a flag instead of as an enum then it should be a list of checkboxes:
For example: `warpAffine` uses `INTER_LINEAR` but it also uses the flags.

File: buildSrc/src/main/java/edu/wpi/gripgenerator/FileParser.java (hunk 1 above)
Comment (JLLeitschuh) [inline buildSrc/src/main/java/edu/wpi/gripgenerator/FileParser.java]: I think its better to make it long and then hide it with ellipses. (ie `...`)

**Comment by JLLeitschuh** [comment]
Problem:
The input arrays and the output array can all have the same or different depths. For example, you
can subtract to 8-bit unsigned arrays and store the difference in a 16-bit signed array. Depth of
the output array is determined by dtype parameter. In the second and third cases above, as well as
in the first case, when src1.depth() == src2.depth(), dtype can be set to the default -1. In this
case the output array will have the same depth as the input array, be it src1, src2 or both.

What if the two array input are not the same type? Then -1 won't work.

**Pull request merged**